src/rabbit_vhost.erl

```diff
@@ -61,37 +65,47 @@ recover(VHost) ->
     ok = rabbit_mirror_queue_misc:on_vhost_up(VHost),
     ok.
 
-%%----------------------------------------------------------------------------
-
--define(INFO_KEYS, [name, tracing, cluster_state]).
+-define(INFO_KEYS, vhost:info_keys()).
 
--spec add(rabbit_types:vhost(), rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
+-spec add(vhost:name(), rabbit_types:username()) -> rabbit_types:ok_or_error(any()).
 
 add(VHost, ActingUser) ->
     case exists(VHost) of
         true  -> ok;
-        false -> do_add(VHost, ActingUser)
+        false -> do_add(VHost, <<"">>, <<"">>, ActingUser)
+    end.
+
+-spec add(vhost:name(), rabbit_types:username(), string(), [atom()]) -> rabbit_types:ok_or_error(any()).
```

Comment:
Should tags be atoms? It would be dangerous and I don't see the need to use atoms here. Strings or binaries would be more practicle IMHO.

Reply:
The order of arguments in the `-spec()` doesn't match the order in the function definition.

Reply:
User tags are atoms.